internal/kibana/agents.go

```diff
@@ -62,62 +62,46 @@ func (c *Client) AssignPolicyToAgent(a Agent, p Policy) error {
 		return fmt.Errorf("could not assign policy to agent; API status code = %d; response body = %s", statusCode, string(respBody))
 	}
 
-	err = c.waitUntilPolicyAssigned(p)
+	err = c.waitUntilPolicyAssigned(a, p)
 	if err != nil {
 		return errors.Wrap(err, "error occurred while waiting for the policy to be assigned to all agents")
 	}
 	return nil
 }
 
-func (c *Client) getTotalAgentForPolicy(p Policy) (int, error) {
-	kuery := url.QueryEscape(fmt.Sprintf("fleet-agents.policy_id:\"%s\"", p.ID))
-	path := fmt.Sprintf("%s/agents?kuery=%s", FleetAPI, kuery)
-	statusCode, respBody, err := c.get(path)
-	if err != nil {
-		return 0, errors.Wrapf(err, "could not check agent status; API status code = %d; policy ID = %s; response body = %s", statusCode, p.ID, string(respBody))
-	}
-	var resp struct {
-		Total int `json:"total"`
-	}
+func (c *Client) waitUntilPolicyAssigned(a Agent, p Policy) error {
+	for {
+		agent, err := c.getAgent(a.ID)
+		if err != nil {
+			return errors.Wrap(err, "can't get the agent")
+		}
 
-	if err := json.Unmarshal(respBody, &resp); err != nil {
-		return 0, errors.Wrap(err, "could not convert agent list (response) to JSON")
+		if agent.PolicyID == p.ID && agent.PolicyRevision == p.Revision {
+			logger.Debugf("Policy revision assigned to the agent (ID: %s)...", a.ID)
+			break
+		}
+
+		logger.Debugf("Wait until the policy (ID: %s, revision: %d) is assigned to the agent (ID: %s)...", p.ID, p.Revision, a.ID)
+		time.Sleep(2 * time.Second)
 	}
-	return resp.Total, nil
+	return nil
 }
 
-func (c *Client) waitUntilPolicyAssigned(p Policy) error {
-	totalAgents, err := c.getTotalAgentForPolicy(p)
+func (c *Client) getAgent(agentID string) (*Agent, error) {
+	statusCode, respBody, err := c.get(fmt.Sprintf("%s/agents/%s", FleetAPI, agentID))
```

Comment:
Does this API exist in older versions of Kibana? IOW, just wondering if this change is backwards compatible.

Reply:
I think it was already there, we just used a different API.

Reply:
I quickly tested this PR with 7.11.0 and it works. So we're good on backwards compatibility.